python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -562,16 +541,23 @@ def _gaussianizeAndMeasure(self, measRecord: lsst.afw.table.SourceRecord,
         # scaling factors and sigmas.
         if measRecord.getFootprint().getArea() == 0:
             self._setFlag(measRecord, self.name, "no_pixel")
-            raise NoPixelError
+            self._setScalingAndSigmaFlags(measRecord, self.config.scalingFactors)
+            raise NoPixelError("No good pixels in footprint", 1)
 
         if (psf := exposure.getPsf()) is None:
             raise measBase.FatalAlgorithmError("No PSF in exposure")
 
         psfSigma = psf.computeShape(center).getTraceRadius()
         if not (psfSigma > 0):  # This captures NaN and negative values.
-            errorCollection = {str(scalingFactor): measBase.MeasurementError("PSF size could not be measured")
-                               for scalingFactor in self.config.scalingFactor}
-            raise GaapConvolutionError(errorCollection)
+            center = measRecord.getCentroid()
+            self.log.debug("Failed to solve for PSF matching kernel in GAaP for (%f, %f): %s",
+                           center.getX(), center.getY(), "GAaP Convolution Error")
```

Comment:
Why does the error message say fail to solve for kernel when we don't even attempt solving for it? It made sense when we were handling this in `fail` but we can be more specific here.